pkgs/by-name/ap/apostrophe/package.nix

```diff
@@ -1,101 +1,115 @@
 {
   lib,
-  stdenv,
   fetchFromGitLab,
+  gtksourceview5,
+  libspelling,
+  fetchFromGitHub,
+  python3Packages,
   meson,
   ninja,
-  wrapGAppsHook3,
   pkg-config,
+  wrapGAppsHook4,
   desktop-file-utils,
-  appstream-glib,
-  python3Packages,
-  glib,
   gobject-introspection,
-  gtk3,
-  webkitgtk,
-  glib-networking,
-  gnome,
-  gspell,
+  libadwaita,
+  webkitgtk_6_0,
   texliveMedium,
   shared-mime-info,
-  libhandy,
-  fira,
-  sassc,
 }:
 
 let
-  pythonEnv = python3Packages.python.withPackages (
-    p: with p; [
-      regex
-      setuptools
-      levenshtein
-      pyenchant
-      pygobject3
-      pycairo
-      pypandoc
-      chardet
-    ]
-  );
-in
-stdenv.mkDerivation rec {
-  pname = "apostrophe";
-  version = "2.6.3";
+  version = "3.0";
 
   src = fetchFromGitLab {
     owner = "World";
-    repo = pname;
+    repo = "apostrophe";
     domain = "gitlab.gnome.org";
     rev = "v${version}";
-    sha256 = "sha256-RBrrG1TO810LidIelYGNaK7PjDq84D0cA8VcMojAW3M=";
+    sha256 = "sha256-wKxRCU00nSk7F8IZNWoLRtGs3m6ol3UBnArtppUOz/g=";
   };
 
+  # Patches are required by upstream. Without the patches
+  # typing `- aaa`, newline, `- bbb` the program crashes
+  gtksourceview5-patched = gtksourceview5.overrideAttrs (prev: {
+    patches = (prev.patches or [ ]) ++ [ "${src}/build-aux/flatpak/sourceview_text_commits.patch" ];
+  });
+
+  libspelling-patched =
+    (libspelling.override { gtksourceview5 = gtksourceview5-patched; }).overrideAttrs
+      (prev: {
+        patches = (prev.patches or [ ]) ++ [ "${src}/build-aux/flatpak/libspelling_text_commits.patch" ];
+      });
+
+  reveal-js = fetchFromGitHub {
+    owner = "hakimel";
+    repo = "reveal.js";
+
+    # keep in sync with upstream shipped version
+    # in build-aux/flatpak/org.gnome.gitlab.somas.Apostrophe.json
+    rev = "4.6.0";
+    hash = "sha256-a+J+GasFmRvu5cJ1GLXscoJ+owzFXsLhCbeDbYChkyQ=";
+  };
+in
+python3Packages.buildPythonApplication rec {
+  inherit version src;
+  pname = "apostrophe";
+  pyproject = false;
+
+  postPatch =
+    ''
+      substituteInPlace build-aux/meson_post_install.py \
```

Comment:
https://github.com/NixOS/nixpkgs/issues/302605

Reply:
Okay I'll tell the developers